src/ConstructiveSolidGeometry/SurfacePrimitives/RegularPrismMantle.jl

```diff
@@ -0,0 +1,55 @@
+struct RegularPrismMantle{N,T,TZ} <: AbstractSurfacePrimitive{T}
+    r::T
+    z::TZ
+    function RegularPrismMantle( N::Integer,
+                   ::Type{T},
+                   r::T,
+                   z::Union{T, <:AbstractInterval{T}}) where {T}
+        @assert typeof(N) <: Integer && N >= 3 "Failed constructing RegularPrismMantle with $(N) points in the polygon base (needs at least 3 points)."
+        new{N,T,typeof(z)}(r, z)
+    end
+end
+
+# Convenience functions
+const TriangularPrismMantle{T,TR} = RegularPrismMantle{3,T,TR}
+const SquarePrismMantle{T,TR}  = RegularPrismMantle{4,T,TR}
+const PentagonalPrismMantle{T,TR} = RegularPrismMantle{5,T,TR}
+const HexagonalPrismMantle{T,TR}  = RegularPrismMantle{6,T,TR}
+
+TriangularPrismMantle(args...) = RegularPrismMantle(3, args...)
+SquarePrismMantle(args...)  = RegularPrismMantle(4, args...)
+PentagonalPrismMantle(args...) = RegularPrismMantle(5, args...)
+HexagonalPrismMantle(args...)  = RegularPrismMantle(6, args...)
+
+#Constructors
+RegularPrismMantle(p::RegularPrism{N,T}; r = 1) where {N,T} = RegularPrismMantle(N, T, T(r), p.z)
+
+function RegularPrismMantle(N::Integer; r = 1, zMin = -1, zMax = 1)
+    T = float(promote_type(typeof.((r, zMin, zMax))...))
+    z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
+    RegularPrismMantle(N, T, T(r), z)
+end
+RegularPrismMantle(N::Integer, r, zMin, zMax) = RegularPrismMantle(N; r = r, zMin = zMin, zMax = zMax)
+
+function RegularPrismMantle(N::Integer, r::R, height::H) where {R<:Real, H<:Real}
+    T = float(promote_type(R,H))
+    RegularPrismMantle( N, T, T(r), T(height/2))
+end
```

Comment:
Do we need these convenience functions (i.e. do we expect the user to define its own `RegularPrismMantle` from scratch?

Reply:
We have them for all the other surface primitives. Its also good for testing